Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -422,6 +424,25 @@ NonnullRefPtr<GUI::Action> HackStudioWidget::create_new_project_action()
     });
 }
 
+NonnullRefPtr<GUI::Action> HackStudioWidget::create_rename_action()
+{
+    return GUI::Action::create("Rename", { Mod_Alt, Key_Return }, [this](const GUI::Action&) {
+        auto files = selected_file_names();
+        if (files.is_empty() || files.size() > 1)
+            return;
+
+        auto pre_filename = files[0];
+
+        String new_name;
+        if (GUI::InputBox::show(window(), new_name, "Enter the new file name:", "Rename file") != GUI::InputBox::ExecOK) {
```

Comment:
It looks like the way this is written, you have to supply the entire path in the input box? I think it would probably be more natural to only rename the file relative to its parent.

i.e. If I have `/home/anon/MyProject/foo.cpp`, I would expect that typing `bar.cpp` into the rename dialog would rename it to `/home/anon/MyProject/bar.cpp`, rather than `/path/where/I/opened/hackstudio/from/bar.cpp`. 

Or, maybe the dialog could fill in the input box with `/home/anon/MyProject/foo.cpp` already and all you have to do is edit the string in place.

`LexicalPath::relative_path` would probably help for the first suggestion.

Reply:
Yup, trying to rename any file within a directory in the tree would fail as rename expects the full relative path. Went ahead and fixed that with some string hacking.